### ISSUE TITLE
transformBundle plugin hook should pass format option

### ISSUE THREAD

**Issue by futurist** [rollup/rollup#867]
I'm doing this plugin:

https://github.com/futurist/rollup-plugin-es3

using transformBundle plugin hook, but want to only transform `cjs` format, seems it's lack the option from plugin to check which format the code from. Maybe:

```js
    transformBundle: function (code, [format??] ) {
      if(format=='cjs');  //do some thing
    }
```

Any idea?

**Comment by thgh** [comment]
Not sure if there is a cleaner way than this:
```js
  let format = 'es'
  return {
    ongenerate: function (bundleOptions) {
      format = bundleOptions.format
    }
    transformBundle: function (code, [format??] ) {
      if(format === 'cjs');  //do some thing
    }
  }
```

**Comment by futurist** [comment]
It's not work, since the event order is `transformBundle` => `ongenerate`

**Comment by thgh** [comment]
The options callback should work then :-)

**Comment by futurist** [comment]
It's not work, when the `bundleOptions` is :
```
{ entry: 'source.js',
  plugins: [ { name: 'es3', options: [Function], transformBundle: [Function] } ] }
```

With the source code:

```
rollup({
      entry: file('files/source.js'),
      plugins: [ es3() ]
    }).then(bundle => {
      const result = bundle.generate({
        format: 'cjs'
      }))
```

**Comment by thgh** [comment]
You could pass the format option to the rollupOptions.

When using the command line, the rollupOptions are also passed to bundle.generate, so in that case you would only have to pass it once.

**Comment by Rich-Harris** [comment]
I can see why this would be useful – intercepting the format in the `options` hook probably works in some cases but is definitely a kludge (wouldn't work with `targets`, for example). Would gladly accept a PR that added a `{ format }` object as the second argument to `transformBundle` 😀

**Comment by Rich-Harris** [comment]
`{format}` is available in 0.34.13 thanks to @tivac 🎉
